lib/modules/metadata.mjs

```diff
@@ -25,7 +25,29 @@ export async function validateDate (doc, { mode = MODES.NORMAL } = {}) {
 
   switch (doc.type) {
     case 'txt': {
-      // TODO: Text type validation
+      const docDate = doc.data.header.date
+
+      if (!docDate || !Object.keys(docDate).length) {
+        result.push(new ValidationWarning('MISSING_DOC_DATE', 'The document date could not be determined.', {
+          ref: 'https://authors.ietf.org/en/rfcxml-vocabulary#date'
+        }))
+      } else {
+        const dt = DateTime.fromObject({
+          year: docDate.year || today.year,
+          month: docDate.month ? DateTime.fromFormat(docDate.month, 'MMMM').month : today.month,
+          day: docDate.day || today.day
+        })
+        const daysDiff = Math.round(dt.diffNow().as('days'))
+        if (daysDiff < -3) {
```

Comment:
The 3 here and on L45 would be better as a named constant (a configuration knob)